test/k8sT/DatapathConfiguration.go

```diff
@@ -481,6 +481,69 @@ var _ = Describe("K8sDatapathConfig", func() {
 		})
 	})
 
+	SkipContextIf(helpers.DoesNotRunOnNetNextKernel, "Wireguard encryption", func() {
+		It("Pod2pod is encrypted", func() {
+			deploymentManager.DeployCilium(map[string]string{
+				"tunnel":               "disabled",
+				"autoDirectNodeRoutes": "true",
+				"wireguard.enabled":    "true",
+				"l7Proxy":              "false",
+			}, DeployCiliumOptionsAndDNS)
+
+			randomNamespace := deploymentManager.DeployRandomNamespaceShared(DemoDaemonSet)
+			deploymentManager.WaitUntilReady()
+
+			k8s1NodeName, k8s1IP := kubectl.GetNodeInfo(helpers.K8s1)
+			k8s2NodeName, k8s2IP := kubectl.GetNodeInfo(helpers.K8s2)
+
+			// Fetch srcPod (testDSClient@k8s1)
+			srcPod, srcPodJSON := fetchPodsWithOffset(kubectl, randomNamespace, "client", "zgroup=testDSClient", k8s2IP, true, 0)
+			srcPodIP, err := srcPodJSON.Filter("{.status.podIP}")
+			Expect(err).Should(BeNil(), "Failure to retrieve pod IP %s", srcPod)
+			srcHost, err := srcPodJSON.Filter("{.status.hostIP}")
+			Expect(err).Should(BeNil(), "Failure to retrieve host of pod %s", srcPod)
+			// Sanity check
+			Expect(srcHost.String()).Should(Equal(k8s1IP))
+
+			// Fetch dstPod (testDS@k8s2)
+			dstPod, dstPodJSON := fetchPodsWithOffset(kubectl, randomNamespace, "server", "zgroup=testDS", k8s1IP, true, 0)
+			dstPodIP, err := dstPodJSON.Filter("{.status.podIP}")
+			Expect(err).Should(BeNil(), "Failure to retrieve IP of pod %s", dstPod)
+			dstHost, err := dstPodJSON.Filter("{.status.hostIP}")
+			Expect(err).Should(BeNil(), "Failure to retrieve host of pod %s", dstPod)
+			// Sanity check
+			Expect(dstHost.String()).Should(Equal(k8s2IP))
+
+			privateIface, err := kubectl.GetPrivateIface()
+			Expect(err).Should(BeNil(), "Cannot determine private iface")
+			cmd := fmt.Sprintf("tcpdump -i %s --immediate-mode -n 'host %s and host %s' -c 1", privateIface, srcPodIP, dstPodIP)
+
+			res1, cancel1, err := kubectl.ExecInHostNetNSInBackground(context.TODO(), k8s1NodeName, cmd)
+			Expect(err).Should(BeNil(), "Cannot exec tcpdump in bg")
+
+			res2, cancel2, err := kubectl.ExecInHostNetNSInBackground(context.TODO(), k8s2NodeName, cmd)
+			Expect(err).Should(BeNil(), "Cannot exec tcpdump in bg")
+
+			// HTTP connectivity test (pod2pod)
+			kubectl.ExecPodCmd(randomNamespace, srcPod,
+				helpers.CurlFail("http://%s:80/", dstPodIP)).ExpectSuccess("Failed to curl dst pod")
+
+			// Check that no unencrypted pod2pod traffic was captured on the direct routing device
+			cancel1()
+			cancel2()
+			Expect(res1.CombineOutput().String()).Should(Not(ContainSubstring("1 packet captured")))
+			Expect(res2.CombineOutput().String()).Should(Not(ContainSubstring("1 packet captured")))
```

Comment:
nit: Shouldn't we check that it does contains `0 packets captured`? Do we expect false positives?

Reply:
I'm afraid that this might introduce flakiness, as I expect the cmd termination not always to catch all stderr output which contains `0 packets captured`. In any case, we run tcpdump with `-c 1` which means terminate immediately after receiving a packet which matches a filter. So if we leak the traffic, then tcpdump should return immediately with the `1 packets captured` output.